core/modules/qdisp/Executive.cc

```diff
@@ -357,103 +275,57 @@ struct printMapEntry {
 };
 
 int Executive::getNumInflight() {
-    std::unique_lock<std::mutex> lock(_requestersMutex);
-    return _requesters.size();
+    std::unique_lock<std::mutex> lock(_incompleteJobsMutex);
+    return _incompleteJobs.size();
 }
 
 std::string Executive::getProgressDesc() const {
     std::ostringstream os;
     {
-        std::lock_guard<std::mutex> lock(_statusesMutex);
-        std::for_each(_statuses.begin(), _statuses.end(), printMapEntry(os, "\n"));
+        std::lock_guard<std::mutex> lock(_jobsMutex);
+        std::for_each(_jobMap.begin(), _jobMap.end(), printMapEntry(os, "\n"));
```

Comment:
If this is the only use of printMapEntry, consider dispensing with helper and rolling it directly in here as a lambda?

```c++
std::ostringstream os;
bool first = true;
{
    std::lock_guard<std::mutex> lock(_jobsMutex);
    std::for_each(_jobMap.begin(), _jobMap.end(), [&](Executive::JobMap::value_type const& entry){
        JobQuery::Ptr job = entry.second;
        if(!first) { os << "\n"; }
        os << "Ref=" << entry.first << " " << job;
        first = false;
    }));
}
```